_test/tests/inc/Utils/PageResolverTest.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace tests\inc\Utils;
+
+use dokuwiki\Utils\PageResolver;
+
+/**
+ * @todo tests that make use of revisions might be wanted
+ */
+class PageResolverTest extends \DokuWikiTest
+{
+    /**
+     * @return \Generator|array
+     * @see testResolveID
+     */
+    public function provideResolveData()
+    {
+        $data = [
+            // relative current in root
+            ['context', 'page', 'page'],
+            ['context', '.page', 'page'],
+            ['context', '.:page', 'page'],
+
+            // relative current in namespace
+            ['lev1:lev2:context', 'page', 'lev1:lev2:page'],
+            ['lev1:lev2:context', '.page', 'lev1:lev2:page'],
+            ['lev1:lev2:context', '.:page', 'lev1:lev2:page'],
+
+            // relative upper in root
+            ['context', '..page', 'page'],
+            ['context', '..:page', 'page'],
+
+            // relative upper in namespace
+            ['lev1:lev2:context', '..page', 'lev1:page'],
+            ['lev1:lev2:context', '..:page', 'lev1:page'],
+            ['lev1:lev2:context', '..:..:page', 'page'],
+            ['lev1:lev2:context', '..:..:..:page', 'page'],
+
+            // deeper nesting
+            ['lev1:lev2:lev3:context', '..page', 'lev1:lev2:page'],
+            ['lev1:lev2:lev3:context', '..:page', 'lev1:lev2:page'],
+            ['lev1:lev2:lev3:context', '..:..page', 'lev1:page'],
+            ['lev1:lev2:lev3:context', '..:..:page', 'lev1:page'],
+            ['lev1:lev2:lev3:context', '..:..:..page', 'page'],
+            ['lev1:lev2:lev3:context', '..:..:..:page', 'page'],
+            ['lev1:lev2:lev3:context', '..:..:..:..page', 'page'],
+            ['lev1:lev2:lev3:context', '..:..:..:..:page', 'page'],
+
+            // strange and broken ones
+            ['lev1:lev2:context', '....:....:page', 'lev1:lev2:page'],
+            ['lev1:lev2:context', '..:..:lev3:page', 'lev3:page'],
+            ['lev1:lev2:context', '..:..:lev3:..:page', 'page'],
+            ['lev1:lev2:context', '..:..:lev3:..:page:....:...', 'page'],
+
+            // relative to current page
+            ['context', '~page', 'context:page'],
+            ['context', '~:page', 'context:page'],
+            ['lev1:lev2:context', '~page', 'lev1:lev2:context:page'],
+            ['lev1:lev2:context', '~:page', 'lev1:lev2:context:page'],
+
+            // start pages
+            ['context', '.:', 'start'],
+            ['foo:context', '.:', 'foo:start'],
+            ['context', 'foo:', 'foo:start'],
+            ['foo:context', 'foo:', 'foo:start'],
+            ['context', '~foo:', 'context:foo:start'],
+            ['foo:context', '~foo:', 'foo:context:foo:start'],
+
+            // empty page links to itself
+            ['context', '', 'context'],
+            ['foo:context', '', 'foo:context'],
+        ];
```

Comment:
Just a note: no examples of autoplural are in the tests at the moment.

Reply:
you're right. I added a test